Dockerfile

```diff
@@ -25,7 +25,7 @@ ADD root/tmp/setup/oci8-extension.sh /tmp/setup/
 RUN chmod 777 /tmp && chmod +t /tmp && \
     /tmp/setup/oci8-extension.sh
 
-RUN mkdir /var/www/moodledata && chown www-data /var/www/moodledata && \
-    mkdir /var/www/phpunitdata && chown www-data /var/www/phpunitdata && \
-    mkdir /var/www/behatdata && chown www-data /var/www/behatdata && \
-    mkdir /var/www/behatfaildumps && chown www-data /var/www/behatfaildumps
+# Set the custom entrypoint up.
+COPY moodle-php-entrypoint /usr/local/bin/
+ENTRYPOINT ["moodle-php-entrypoint"]
```

Comment:
I'd go for the full path here

Reply:
Besides, why not using the `.sh` extension here?

Reply:
I copied the same style that the official PHP image uses: https://github.com/docker-library/php/blob/master/7.1/stretch/apache/Dockerfile#L258-L260
I wanted to stay consistent with that.

Reply:
Understood.

In other pending issues in other docker-verse areas  (e.g. https://github.com/moodlehq/moodle-docker/pull/90) we - I - recommended to use the full path, that's the reason why I was wondering why not here.
We should being consistent with Moodle images regardless of using an upstream image which adopts different conventions or we should define how to deal with conventions even in Moodle images e.g. adopting the one for each upstream image - not solid IMHO, for consistency about how to code the Moodle images, including if one day the upstream image will be different.

The real question is, to be "consistent" why not removing `.sh` from the other scripts used here in this image too?
BTW not really important, just to understand the rationale behind that coding style.

OT: I'm fun of that extension since it allows a sysadm to quickly understand if it is a binary vs a script w/o opening that file or using `file <filename>`.

HTH,
Matteo

Reply:
> In other pending issues in other docker-verse areas (e.g. [moodlehq/moodle-docker#90](https://github.com/moodlehq/moodle-docker/pull/90)) we - I - recommended to use the full path, that's the reason why I was wondering why not here

That's a different use-case and meaning. That is making use of content within the image where the $PATH cannot be confirmed. We cannot guarantee that `python` will be in the $PATH depending on the user that apache is run as in all circumstances. We use the full path there because it is an external tool making use of something within the image.

>  We should being consistent with Moodle images regardless of using an upstream image which adopts different conventions or we should define how to deal with conventions even in Moodle images e.g. adopting the one for each upstream image - not solid IMHO, for consistency about how to code the Moodle images, including if one day the upstream image will be different.

I am trying to move these images to consistency. To do so I am planning to switch the build process to make use of `bashbrew`, and follow the guidelines laid out in the official docker images. I have opened #45 to investigate how this can be achieved.

> The real question is, to be "consistent" why not removing `.sh` from the other scripts used here in this image too?
> BTW not really important, just to understand the rationale behind that coding style.

Because I plan to entirely remove those scripts.  As per https://github.com/docker-library/official-images/blob/master/README.md#clarity they should be entirely part of the Dockerfile.
However the change to doing so is non-trivial. I expect to move the existing Dockerfiles to use a machine-generated Dockerfile as part of that change, and to switch build process to make use of `bashbrew` and build on our own infrastructure at the same time.

> OT: I'm fun of that extension since it allows a sysadm to quickly understand if it is a binary vs a script w/o opening that file or using `file <filename>`.

sysadmins shouldn't care. It's in a bin folder and it is executable. It could be sh, python, or compiled code. The important thing is that it executes. This is also a docker entrypoint and not something we expect sysadmins to be playing with that much. Remember, this isn't an image intended for serving Moodle. This is an image intended solely for _Developer_  usage.